nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -461,179 +162,311 @@ def forward(self, obj):
+    It can be initialate with constant argument, and positions of each non-constant
+    argument. When called, it can put arguments into correct position, then call the
+    function.
 
+    Attributes
+    ----------
+    func: Callable
+        The function or method to be called.
+    positional: List[Any]
+        Positional arguments values. The placeholder is None if it's non-constant.
+    keyword: Dict[str, Any]
+        Keyword arguments values. . The placeholder is None if it's non-constant.
+    undetermined: List[int | str]
+        A list of the right positions of arguments.
+        Position is an int in positional or a str in keyword.
+    special_treat: Dict[int | str, Callable]
+        A Dict of the positions and methods.
+        The values of these positions should be treat by those methods.
 
-def upsample_bilinear2d_python(node, speedup):
-    class UpsampleModule(torch.nn.Module):
-        def __init__(self, size_list, scale_list):
-            super(UpsampleModule, self).__init__()
-            self.size_list = size_list
-            self.scale_list = scale_list
-
-        def forward(self, *args):
-            """
-            The first input of args is the target tensor to upsample
-            , the following parameters is useless, because we already
-            get the size_list and the scale_list by parsing the cpp_nodes.
-            """
-            return torch.nn.functional.upsample_bilinear(args[0],
-                                                         size=self.size_list, scale_factor=self.scale_list)
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[3].node()
-    size_list = None
-    scale_list = None
-
-    if size_list_node.kind() == 'prim::ListConstruct':
-        size_list = translate_list(inputs[1], speedup)
-    if scale_list_node.kind() == 'prim::ListConstruct':
-        scale_list = translate_list(inputs[3], speedup)
-    return UpsampleModule(size_list, scale_list)
-
-
-def upsample_nearest2d_python(node, speedup):
-    class UpsampleModule(torch.nn.Module):
-        def __init__(self, size_list, scale_list):
-            super(UpsampleModule, self).__init__()
-            self.size_list = size_list
-            self.scale_list = scale_list
-
-        def forward(self, *args):
-            """
-            The first input of args is the target tensor to upsample
-            , the following parameters is useless, because we already
-            get the size_list and the scale_list by parsing the cpp_nodes.
-            """
-            return torch.nn.functional.upsample_nearest(args[0],
-                                                        size=self.size_list, scale_factor=self.scale_list)
-    c_node = node.key_node
-    inputs = list(c_node.inputs())
-    size_list_node = inputs[1].node()
-    scale_list_node = inputs[2].node()
-    size_list = None
-    scale_list = None
+    """
+
+    func: Callable
```

Comment:
This type hints should be written inside `__init__()`.
Outside `__init__()` they are defining static fields.